src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient.cs

```diff
@@ -34,128 +35,177 @@ public ImageStoreClient(ServiceFabricHttpClient httpClient)
-            serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
-            var requestId = Guid.NewGuid().ToString();
-            var url = "ImageStore/{contentPath}";
-            url = url.Replace("{contentPath}", Uri.EscapeDataString(contentPath));
-            var queryParams = new List<string>();
-            
-            // Append to queryParams if not null.
-            serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
-            queryParams.Add("api-version=6.2");
-            url += "?" + string.Join("&", queryParams);
-            
-            HttpRequestMessage RequestFunc()
+
+            var store = await GetImageStoreConnectionString();
+            if (IsLocalStore(store))
             {
-                var request = new HttpRequestMessage()
-                {
-                    Method = HttpMethod.Get,
-                };
-                return request;
+                var path = Path.Combine(imageStorePath, contentPath);
+                var directories = Directory.EnumerateDirectories(path, "*", SearchOption.AllDirectories).Select(d => new FolderInfo(d, (long)Directory.EnumerateFiles(d).Count())).ToList();
+                var files = Directory.EnumerateFiles(path, "*", SearchOption.AllDirectories).Select(f => new System.IO.FileInfo(f)).Select(f => new Common.FileInfo(f.Length.ToString(), new FileVersion(), f.LastWriteTimeUtc, f.FullName.Replace(imageStorePath, ""))).ToList();
+                return new ImageStoreContent(files, directories);
             }
+            else
+            {
+                serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
+                var requestId = Guid.NewGuid().ToString();
+                var url = "ImageStore/{contentPath}";
+                url = url.Replace("{contentPath}", Uri.EscapeDataString(contentPath));
+                var queryParams = new List<string>();
+
+                // Append to queryParams if not null.
+                serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
+                queryParams.Add("api-version=6.2");
+                url += "?" + string.Join("&", queryParams);
+
+                HttpRequestMessage RequestFunc()
+                {
+                    var request = new HttpRequestMessage()
+                    {
+                        Method = HttpMethod.Get,
+                    };
+                    return request;
+                }
 
-            return this.httpClient.SendAsyncGetResponse(RequestFunc, url, ImageStoreContentConverter.Deserialize, requestId, cancellationToken);
+                return await this.httpClient.SendAsyncGetResponse(RequestFunc, url, ImageStoreContentConverter.Deserialize, requestId, cancellationToken);
+            }
         }
 
         /// <inheritdoc />
-        public Task DeleteImageStoreContentAsync(
+        public async Task DeleteImageStoreContentAsync(
```

Comment:
Directly returning Task is better

Reply:
Since this method now has to retrieve the connection string, it can no longer just return a task. I also see no technical reason why returning a task directly is better, you'd have to get pretty deep into a micro benchmark for that to show up, and in this instance I believe readability takes precedence